### PR TITLE
Fix lang issue on zypper absent

### LINKED ISSUE #1168: zypper remove ignored which LANG=fr_FR
##### SUMMARY
The `zypper` module does not remove packages with `state: absent` when the local is set something other than 'en_US'.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- plugins/modules/packaging/os/zypper.py

##### ANSIBLE VERSION
```paste below
ansible 2.9.14
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/xenlo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.6 (default, Sep 25 2020, 00:00:00) [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

##### CONFIGURATION

```paste below
DEFAULT_STDOUT_CALLBACK(/etc/ansible/ansible.cfg) = yaml
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /home/xenlo/.ansible/vault_keyring
```

##### OS / ENVIRONMENT
*Controler*:
- Fedora 32
- LANG: fr_BE.UTF-8

*Target host*:
- SLES 15
- LANG: POSIX

##### STEPS TO REPRODUCE

On a target SLES (or openSUSE) host ensure that a rpm package is installed with zypper:
```shell
target:~ # zypper install libwbclient0
[…]
target:~ # zypper search libwbclient0
[…]
S | Nom          | Résumé                    | Type  
--+--------------+---------------------------+-------
i | libwbclient0 | Samba libwbclient Library | paquet
```

And from there try to remove from an Ansible controle with an other LANG than 'en_US' (either with `ansible` ad-hoc or `ansible-playbook):
```yaml
controler:~/Projects/344$ LANG=fr_FR.UTF8  ansible -i inventories/ target -m zypper -a "name=libwbclient0 state=absent" -u root
target | SUCCESS => {​​​​​​​
    "changed": false,
    "name": [
        "libwbclient0"
    ],
    "rc": 0,
    "state": "absent",
    "update_cache": false
}​​​​​​​
```
You get `"changed": false,` and `"state": "absent",`.

##### EXPECTED RESULTS
I expect to have the package removed from the target and as output "changed: true". Just like if I run the same ansible command but with 'en_US'. 

```shell
controler:~/Projects/344$ LANG=en_US.UTF8 ansible -i inventories/ target -m zypper -a "name=colordiff state=absent" -u root
target | CHANGED => {
    "changed": true,
    "cmd": [
        "/usr/bin/zypper",
        "--quiet",
        "--non-interactive",
        "--xmlout",
        "remove",
        "--type",
        "package",
        "colordiff"
    ],
    "name": [
        "colordiff"
    ],
    "rc": 0,
    "state": "absent",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "<?xml version='1.0'?>\n<stream>\n<install-summary download-size=\"0\" space-usage-diff=\"-54086\">\n<to-remove>\n<solvable type=\"package\" name=\"colordiff\" edition=\"1.0.18-bp150.2.4\" arch=\"noarch\" summary=\"Colour-highlighted &apos;diff&apos; output\">\n<description>The Perl script colordiff is a wrapper for &apos;diff&apos; and produces the same\noutput but with pretty &apos;syntax&apos; highlighting. Colour schemes can be\ncustomized.</description></solvable>\n</to-remove>\n</install-summary>\n<prompt id=\"0\">\n<text>Continue?</text>\n<option default=\"1\" value=\"y\" desc=\"Yes, accept the summary and proceed with installation/removal of packages.\"/>\n<option value=\"n\" desc=\"No, cancel the operation.\"/>\n<option value=\"v\" desc=\"Toggle display of package versions.\"/>\n<option value=\"a\" desc=\"Toggle display of package architectures.\"/>\n<option value=\"r\" desc=\"Toggle display of repositories from which the packages will be installed.\"/>\n<option value=\"m\" desc=\"Toggle display of package vendor names.\"/>\n<option value=\"d\" desc=\"Toggle between showing all details and as few details as possible.\"/>\n<option value=\"g\" desc=\"View the summary in pager.\"/>\n</prompt>\n</stream>\n",
    "stdout_lines": [
        "<?xml version='1.0'?>",
        "<stream>",
        "<install-summary download-size=\"0\" space-usage-diff=\"-54086\">",
        "<to-remove>",
        "<solvable type=\"package\" name=\"colordiff\" edition=\"1.0.18-bp150.2.4\" arch=\"noarch\" summary=\"Colour-highlighted &apos;diff&apos; output\">",
        "<description>The Perl script colordiff is a wrapper for &apos;diff&apos; and produces the same",
        "output but with pretty &apos;syntax&apos; highlighting. Colour schemes can be",
        "customized.</description></solvable>",
        "</to-remove>",
        "</install-summary>",
        "<prompt id=\"0\">",
        "<text>Continue?</text>",
        "<option default=\"1\" value=\"y\" desc=\"Yes, accept the summary and proceed with installation/removal of packages.\"/>",
        "<option value=\"n\" desc=\"No, cancel the operation.\"/>",
        "<option value=\"v\" desc=\"Toggle display of package versions.\"/>",
        "<option value=\"a\" desc=\"Toggle display of package architectures.\"/>",
        "<option value=\"r\" desc=\"Toggle display of repositories from which the packages will be installed.\"/>",
        "<option value=\"m\" desc=\"Toggle display of package vendor names.\"/>",
        "<option value=\"d\" desc=\"Toggle between showing all details and as few details as possible.\"/>",
        "<option value=\"g\" desc=\"View the summary in pager.\"/>",
        "</prompt>",
        "</stream>"
    ],
    "update_cache": false
```
##### ACTUAL RESULTS
Output with `-vvvv` in [this gist](https://gist.github.com/xenlo/bd49aa660a32b4df9a64fa94f61e3d71).

### PR DESCRIPTION
**Pull request by xenlo**
##### SUMMARY
Fix #1168

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- plugins/modules/packaging/os/zypper.py

### CODE REVIEW

File: changelogs/fragments/1175-zypper-absent-lang.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- zypper - force LANG=C to as zypper is looking in xml output where attribute could be translated (https://github.com/ansible-collections/community.general/issues/1175).
```
Comment (felixfontein) [inline changelogs/fragments/1175-zypper-absent-lang.yml:2]:
```suggestion
- zypper - force ``LANG=C`` to as zypper is looking in XML output where attribute could be translated (https://github.com/ansible-collections/community.general/issues/1175).
```

**Comment by evrardjp** [comment]
Makes sense to me.

**Comment by felixfontein** [comment]
This change should be pretty safe. @xenlo did you test whether it actually solves your problem?

**Comment by xenlo** [comment]
Yes it works as expected. 
Tested on OpenSUSE 15.1 and SLES 15 boxes:
```shell
(344) xenlo@indigo-faraday:~/Projects/344$ LANG=fr_FR.UTF8  ansible -i inventories/ target_os151 -m zypper -a "name=colordiff state=absent" -u root
target_os151 | CHANGED => {
    "changed": true,
    "cmd": [
        "/usr/bin/zypper",
        "--quiet",
        "--non-interactive",
        "--xmlout",
        "remove",
        "--type",
        "package",
        "colordiff"
    ],
    "name": [
        "colordiff"
    ],
    "rc": 0,
    "state": "absent",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "<?xml version='1.0'?>\n<stream>\n<install-summary download-size=\"0\" space-usage-diff=\"-54086\" packages-to-change=\"1\">\n<to-remove>\n<solvable type=\"package\" name=\"colordiff\" edition=\"1.0.18-lp151.2.1\" arch=\"noarch\" summary=\"Colour-highlighted &apos;diff&apos; output\">\n<description>The Perl script colordiff is a wrapper for &apos;diff&apos; and produces the same\noutput but with pretty &apos;syntax&apos; highlighting. Colour schemes can be\ncustomized.</description></solvable>\n</to-remove>\n</install-summary>\n<prompt id=\"0\">\n<text>Continue?</text>\n<option default=\"1\" value=\"y\" desc=\"Yes, accept the summary and proceed with installation/removal of packages.\"/>\n<option value=\"n\" desc=\"No, cancel the operation.\"/>\n<option value=\"v\" desc=\"Toggle display of package versions.\"/>\n<option value=\"a\" desc=\"Toggle display of package architectures.\"/>\n<option value=\"r\" desc=\"Toggle display of repositories from which the packages will be installed.\"/>\n<option value=\"m\" desc=\"Toggle display of package vendor names.\"/>\n<option value=\"d\" desc=\"Toggle between showing all details and as few details as possible.\"/>\n<option value=\"g\" desc=\"View the summary in pager.\"/>\n</prompt>\n</stream>\n",
    "stdout_lines": [
        "<?xml version='1.0'?>",
        "<stream>",
        "<install-summary download-size=\"0\" space-usage-diff=\"-54086\" packages-to-change=\"1\">",
        "<to-remove>",
        "<solvable type=\"package\" name=\"colordiff\" edition=\"1.0.18-lp151.2.1\" arch=\"noarch\" summary=\"Colour-highlighted &apos;diff&apos; output\">",
        "<description>The Perl script colordiff is a wrapper for &apos;diff&apos; and produces the same",
        "output but with pretty &apos;syntax&apos; highlighting. Colour schemes can be",
        "customized.</description></solvable>",
        "</to-remove>",
        "</install-summary>",
        "<prompt id=\"0\">",
        "<text>Continue?</text>",
        "<option default=\"1\" value=\"y\" desc=\"Yes, accept the summary and proceed with installation/removal of packages.\"/>",
        "<option value=\"n\" desc=\"No, cancel the operation.\"/>",
        "<option value=\"v\" desc=\"Toggle display of package versions.\"/>",
        "<option value=\"a\" desc=\"Toggle display of package architectures.\"/>",
        "<option value=\"r\" desc=\"Toggle display of repositories from which the packages will be installed.\"/>",
        "<option value=\"m\" desc=\"Toggle display of package vendor names.\"/>",
        "<option value=\"d\" desc=\"Toggle between showing all details and as few details as possible.\"/>",
        "<option value=\"g\" desc=\"View the summary in pager.\"/>",
        "</prompt>",
        "</stream>"
    ],
    "update_cache": false
}
```

**Review by felixfontein — APPROVED** [review]

**Comment by mator** [comment]
shipit

**Comment by mator** [comment]
Well, still wasn't able to reproduce issue with my setup (install) , but if the patch fixes issue, then it is `ok` for me. Thanks.

shipit

**Comment by felixfontein** [comment]
@xenlo thanks for fixing this!
@evrardjp @mator thanks for reviewing and testing!

**Pull request merged**